src/lib/storage/storage_manager.cpp

```diff
@@ -10,38 +10,48 @@
 namespace opossum {
 
 StorageManager& StorageManager::get() {
-  return *(new StorageManager());
-  // A really hacky fix to get the tests to run - replace this with your implementation
+  static StorageManager instance;
+  return instance;
 }
 
 void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> table) {
-  // Implementation goes here
+  _tables.insert_or_assign(name, table);
 }
 
 void StorageManager::drop_table(const std::string& name) {
-  // Implementation goes here
+  auto search = _tables.find(name);
+  DebugAssert(search != _tables.end(), "Tablename could not be found.");
+  _tables.erase(name);
 }
 
 std::shared_ptr<Table> StorageManager::get_table(const std::string& name) const {
-  // Implementation goes here
-  return nullptr;
+  auto search = _tables.find(name);
+  DebugAssert(search != _tables.end(), "Tablename could not be found.");
+  return search->second;
 }
 
 bool StorageManager::has_table(const std::string& name) const {
-  // Implementation goes here
-  return false;
+  auto search = _tables.find(name);
+  return search != _tables.end();
 }
 
 std::vector<std::string> StorageManager::table_names() const {
-  throw std::runtime_error("Implement StorageManager::table_names");
+  std::vector<std::string> names;
```

Comment:
one could use vector::reserve here because the size is known by the size of _tables

Reply:
Fixed in next commit